Content.Server/Fluids/EntitySystems/SpillableSystem.cs

```diff
@@ -1,33 +1,196 @@
+    }
+
+    private void SpillOnLand(EntityUid uid, SpillableComponent component, LandEvent args)
+    {
+        if (args.User != null &&
+            _solutionContainerSystem.TryGetSolution(uid, component.SolutionName, out var solutionComponent))
+        {
+            var solution = _solutionContainerSystem.Drain(uid, solutionComponent, solutionComponent.DrainAvailable);
+            SpillAt(solution, EntityManager.GetComponent<TransformComponent>(uid).Coordinates, "PuddleSmear");
+        }
+    }
+
+    private void AddSpillVerb(EntityUid uid, SpillableComponent component, GetOtherVerbsEvent args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        if (!_solutionContainerSystem.TryGetDrainableSolution(args.Target.Uid, out var solution))
+            return;
+
+        if (solution.DrainAvailable == FixedPoint2.Zero)
+            return;
+
+        Verb verb = new();
+        verb.Text = Loc.GetString("spill-target-verb-get-data-text");
+        // TODO VERB ICONS spill icon? pouring out a glass/beaker?
+        verb.Act = () =>
         {
-            _solutionContainerSystem
-                .Drain(uid, solutionComponent, solutionComponent.DrainAvailable)
-                .SpillAt(EntityManager.GetComponent<TransformComponent>(uid).Coordinates, "PuddleSmear");
+            var puddleSolution = _solutionContainerSystem.SplitSolution(args.Target.Uid,
+                solution, solution.DrainAvailable);
+            SpillAt(puddleSolution, args.Target.Transform.Coordinates, "PuddleSmear");
+        };
+        verb.Impact = LogImpact.Medium; // dangerous reagent reaction are logged separately.
+        args.Verbs.Add(verb);
+    }
+
+    /// <summary>
+    ///     Spills solution at the specified grid coordinates.
+    /// </summary>
+    /// <param name="solution">Initial solution for the prototype.</param>
+    /// <param name="coordinates">The coordinates to spill the solution at.</param>
+    /// <param name="prototype">The prototype to use.</param>
+    /// <param name="overflow">If the puddle overflow will be calculated. Defaults to true.</param>
+    /// <param name="sound">Whether or not to play the spill sound.</param>
+    /// <param name="combine">Whether to attempt to merge with existing puddles</param>
+    /// <returns>The puddle if one was created, null otherwise.</returns>
+    public PuddleComponent? SpillAt(Solution solution, EntityCoordinates coordinates, string prototype,
+        bool overflow = true, bool sound = true, bool combine = true)
+    {
+        if (solution.TotalVolume == 0) return null;
+        
+
+        if (!_mapManager.TryGetGrid(coordinates.GetGridId(EntityManager), out var mapGrid))
+            return null; // Let's not spill to space.
+
+        return SpillAt(mapGrid.GetTileRef(coordinates), solution, prototype, overflow, sound,
+            combine: combine);
```

Comment:
So turns out that `GetTileRef` can still return space-tiles. I'm not actually sure whether the use of `_mapManager.TryGetGrid(coordinates...` prevents that or not. So maybe still add a `if (TileRef.Tile.IsEmpty)` check just to make sure.